xchange-stream-coinjar/src/main/java/info/bitrich/xchangestream/coinjar/CoinjarStreamingAdapters.java

```diff
@@ -0,0 +1,90 @@
+package info.bitrich.xchangestream.coinjar;
+
+import info.bitrich.xchangestream.coinjar.dto.CoinjarWebSocketBookEvent;
+import info.bitrich.xchangestream.coinjar.dto.CoinjarWebSocketOrderEvent;
+import info.bitrich.xchangestream.coinjar.dto.CoinjarWebSocketUserTradeEvent;
+import java.math.BigDecimal;
+import java.sql.Date;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.knowm.xchange.coinjar.CoinjarAdapters;
+import org.knowm.xchange.currency.Currency;
+import org.knowm.xchange.currency.CurrencyPair;
+import org.knowm.xchange.dto.Order;
+import org.knowm.xchange.dto.trade.LimitOrder;
+import org.knowm.xchange.dto.trade.MarketOrder;
+import org.knowm.xchange.dto.trade.UserTrade;
+
+class CoinjarStreamingAdapters {
+
+  public static CurrencyPair adaptTopicToCurrencyPair(String topic) {
+    if (topic.startsWith("book")) {
+      topic = topic.substring(5);
+      Currency base = new Currency(topic.substring(0, 3));
+      Currency counter = new Currency(topic.substring(3, 6));
+      return new CurrencyPair(base, counter);
```

Comment:
does this work with LINK?

Reply:
No, updated this logic to follow coinjar's logic that adds a `-` when either of the assets has four chars.

Reply:
should this code not be updated to use that updated logic here?